blueflood-core/src/main/java/com/rackspacecloud/blueflood/eventemitter/RollupEventEmitter.java

```diff
@@ -41,10 +52,30 @@ private RollupEventEmitter() {
     @Override
     public Future emit(final String event, final RollupEvent... eventPayload) {
         //TODO: This hack will go away after Kafka Serializer is made generic
-        if(eventPayload[0].getRollup() instanceof BasicRollup) {
+        if(eventPayload[0].getRollup() instanceof BasicRollup && super.hasListeners(ROLLUP_EVENT_NAME)) {
             return eventExecutors.submit(new Runnable() {
                 @Override
                 public void run() {
+                    if (Configuration.getInstance().getBooleanProperty(CoreConfig.USE_ES_FOR_UNITS) && Configuration.getInstance().getListProperty(CoreConfig.DISCOVERY_MODULES).contains(Util.ElasticIOPath)) {
+                        QueryDiscoveryModuleLoader.loadDiscoveryModule();
+                        final DiscoveryIO discoveryIO = QueryDiscoveryModuleLoader.getDiscoveryInstance();
+                        // TODO: Sync for now, but we will have to make it async eventually
+                        Lists.transform(Arrays.asList(eventPayload), new Function<RollupEvent, RollupEvent>() {
+                            @Override
+                            public RollupEvent apply(RollupEvent event) {
+                                String unit;
+
```

Comment:
NIT: Space is unnecessary? Possibly we should leave the style comments to check style